source/core/TypeAndDataManager.cpp

```diff
@@ -816,6 +818,9 @@ Boolean TypeCommon::IsA(TypeRef otherType, Boolean compatibleStructure)
 
     if (!bMatch && (otherType->Name().Length() > 0)) {
         bMatch = IsA(otherType);
+        if (_encoding == kEncoding_RefNum) {
+            return otherType->BitEncoding() == kEncoding_RefNum && bMatch && GetSubElement(0)->IsA(otherType->GetSubElement(0), compatibleStructure);
+        }
```

Comment:
I think these 3 lines should be moved as additional "else if" case same as "} else if (thisEncoding == kEncoding_Refnum && otherEncoding == kEncoding_Refnum) {" for better readability unless this has to be here.

Reply:
Please correct me if I am wrong. Also, until we unify this IsA() and the CompareType above, my understanding is that we have to duplicate the checks in both places. One gets used in some cases and another one gets used in some other cases.

Reply:
Is it enough to check only GetSubElement(0)? not necessary to check all all SubElementCount() subElements?

Reply:
This has to be here because it depends on the results of  'bMatch = IsA(otherType);'  I could move the refnum test up with the other if's and duplicate that call there, but I'm not sure if it's cleaner.
CompareType() actually calls IsA in its last section, so it does wind up sharing this bit of code.
Refnums types can only have one subelement type, so it's sufficient.